### PR TITLE
bump bitcoin-0.30 and miniscript-10.0

### PR DESCRIPTION
**Pull request by edouardparis**
`psbt::serialize::{Serialize, Deserialize}` are private traits I re implemented them in order to serialize psbt key/value pairs.
Hopefully one day rust-bitcoin support PSBTv2

### CODE REVIEW

**Comment by edouardparis** [comment]
I want the code to look like exactly like `bitcoin::psbt::serialize` module in order to reduce risk of errors, also rust-bitcoin psbt module exposes publicly only a few methods for few types. In order to have the PSBTv2 transition I needed to have access to a very low level of the PSBT mapping.

**Comment by darosior** [comment]
Oh, gotcha. I overlooked the v2 part.

File: bitcoin_client_rs/tests/client.rs [hunk 1]
```diff
@@ -310,15 +310,50 @@ async fn test_sign_psbt() {
...
+                                    .unwrap(),
+                                hex::encode(sig.to_vec())
+                            );
+                        }
+                        _ => {}
+                    }
+                }
+            }
+        }
+
```
Comment (bigspider) [inline bitcoin_client_rs/tests/client.rs:356]: There's identical code in lines 362:395, perhaps it can be in a function?
Even an inner function might do.

**Review by bigspider — CHANGES_REQUESTED** [review]
Thank you!
Only commented on a couple of nits.

**Review by bigspider — APPROVED** [review]

**Pull request merged**